### ISSUE TITLE
No warning issued if [Obsolete(...)] contains \n

### ISSUE THREAD

**Issue by rainersigwald** [dotnet/roslyn#42172]
_From @schibu007 on Wednesday, March 4, 2020 10:44:55 AM_

There is no warning issued if the [Obsolete(...)] attribute contains a message with **\n** (\n alone, not \r\n).
 It works in VS, however:
![image](https://user-images.githubusercontent.com/45459443/75871695-3da48780-5e0d-11ea-9fcf-e78b5c4530e3.png)

### Steps to reproduce
Project file
```xml
<Project Sdk="Microsoft.NET.Sdk">
  <PropertyGroup>
    <TargetFramework>netstandard2.0</TargetFramework>
  </PropertyGroup>
</Project>
```

```cs
using System;

namespace Lib
{
    [Obsolete("\n")]
    public class Class1 { }

    public class Class2
    {
        public void Method() =>  new Class1();
    }
}
```

Command line
```
msbuild
```
### Expected behavior
    **1 Warning(s)**
    0 Error(s)

### Actual behavior
Build succeeded.
    **0 Warning(s)**
    0 Error(s)

### Environment data
`msbuild /version` output:
```
Microsoft (R) Build Engine version 16.4.0+e901037fe for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

16.4.0.56107
```
OS info: Windows 10 1909

_Copied from original issue: microsoft/msbuild#5156_

**Comment by rainersigwald** [comment]
_From @rainersigwald on Wednesday, March 4, 2020 10:33:58 PM_

It looks like this is actually an error with the Roslyn compiler server. I will move the bug to the appropriate repo.

```sh-session
$ dotnet build -p:UseSharedCompilation=true -t:Rebuild
Microsoft (R) Build Engine version 16.5.0+d4cbfca49 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 32.29 ms for S:\repro\microsoft\msbuild\issues\5156\5156.csproj.
  5156 -> S:\repro\microsoft\msbuild\issues\5156\bin\Debug\netcoreapp3.1\5156.dll

Build succeeded.
    0 Warning(s)
    0 Error(s)

Time Elapsed 00:00:00.96

$ dotnet build -p:UseSharedCompilation=false -t:Rebuild
Microsoft (R) Build Engine version 16.5.0+d4cbfca49 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Restore completed in 32.16 ms for S:\repro\microsoft\msbuild\issues\5156\5156.csproj.
Program.cs(12,35): warning CS0618: 'Program.Class1' is obsolete: ' [S:\repro\microsoft\msbuild\issues\5156\5156.csproj]
  5156 -> S:\repro\microsoft\msbuild\issues\5156\bin\Debug\netcoreapp3.1\5156.dll

Build succeeded.

Program.cs(12,35): warning CS0618: 'Program.Class1' is obsolete: ' [S:\repro\microsoft\msbuild\issues\5156\5156.csproj]
    1 Warning(s)
    0 Error(s)

Time Elapsed 00:00:01.51
```

**Comment by YairHalberstadt** [comment]
Why should the be a warning?

**Comment by schibu007** [comment]
Because Class1 is [Obsolete]. The [Obsolete] warning is missing in that case.

**Comment by YairHalberstadt** [comment]
Oh I see. I misunderstood the problem as VS not warning you that you used `\n` instead of `\r\n` or something.
Thanks for clarifying.

**Comment by jcouv** [comment]
Fixed in 16.8 by @svick. Thanks!
